### DISCUSSION TITLE
Running tracker on multiple videos simultaneously (threading)?

### DISCUSSION THREAD

**Discussion by rsorbello** [mikel-brostrom/yolo_tracking#776] [General]
Hi everyone, I am trying to use this tracking repo on multi videos simultaneously but have been running into issues with the threading.  I have basically reformatted the track.py code, so I can call the same class for each thread being initiated.  Each class will be initiated with a different video.  But for some reason, when I run two threads, it continues to crash on one thread but run smoothly on the other.  Has anybody ever attempted this before and been successful? Is there some sort of blocker in the code that prevents running on multiple threads? Thank you very much in advance.

**Comment by mikel-brostrom** [comment]
Multi-video tracking is already supported.: https://github.com/mikel-brostrom/yolov8_tracking/issues/583#issuecomment-1301791768. No need to multithreading unless you goal is something different here...

**Reply by rsorbello** [reply]
Thank you for the post!

**Comment by rsorbello** [comment]
Hi Mikel, 

Just one quick follow up to this.  I noticed once I add multiple video streams to the code, the processing of frames slows down to less than real time (from ~30 fps to ~10 fps).  I am not sure if this is because of my laptop.  Do you run into this as well? Ideally I would like to be able to process 2 or more video streams in real time.
